### PR TITLE
Uses export time minus 5 minutes for export document version

### PR DESCRIPTION
**Pull request by graytaylor0**
### Description
Changes the export version from being `0` to the `exportTime - 5 minutes`. This is so stream events still get priority over export Events, but new exports to the same index will not get version conflicts with OpenSearch with this change. 
 
Tested with export and streams, where updates/deletes were made after starting the pipeline, and the result was this in OpenSearch

```
Export version
1700033214421000

Update item
1700033604000000
```
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: data-prepper-plugins/dynamodb-source/src/test/java/org/opensearch/dataprepper/plugins/source/dynamodb/export/DataFileLoaderFactoryTest.java [hunk 1]
```diff
@@ -61,6 +62,7 @@ class DataFileLoaderFactoryTest {
     private final String prefix = UUID.randomUUID().toString();
 
     private final String exportArn = tableArn + "/export/01693291918297-bfeccbea";
+    private final String exportTime = "1976-01-01T00:00:00Z";
```
[the remark this answers is not in the document]
Comment (dlvenable) [reply data-prepper-plugins/dynamodb-source/src/test/java/org/opensearch/dataprepper/plugins/source/dynamodb/export/DataFileLoaderFactoryTest.java:65]: Actually, this would be better.

```
Instant.EPOCH.toString()
```

[the remark this answers is not in the document]
Comment (graytaylor0) [reply data-prepper-plugins/dynamodb-source/src/test/java/org/opensearch/dataprepper/plugins/source/dynamodb/export/DataFileLoaderFactoryTest.java:65]: Yeah I was just going for a static value, not necessarily the start of epoch time.

File: data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/export/DataFileScheduler.java [hunk 2]
```diff
@@ -111,9 +111,17 @@ private void processDataFilePartition(DataFilePartition dataFilePartition) {
         if (!acknowledgmentsEnabled) {
             runLoader.whenComplete(completeDataLoader(dataFilePartition));
         } else {
-            runLoader.whenComplete((v, ex) -> numOfWorkers.decrementAndGet());
+            runLoader.whenComplete((v, ex) -> {
+                numOfWorkers.decrementAndGet();
+                if (numOfWorkers.get() == 0) {
+                    activeExportS3ObjectConsumersGauge.decrementAndGet();
```
[the remark this answers is not in the document]
Comment (graytaylor0) [reply data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/export/DataFileScheduler.java:117]: These are only the same value because we have the MAX_JOB_COUNT as 1, but I suppose that is unlikely to change again

[the remark this answers is not in the document]
Comment (graytaylor0) [reply data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/export/DataFileScheduler.java:117]: But the goal of this metric is to show how many nodes are processing at least 1 data file

**Review by dlvenable — APPROVED** [review]

**Review by dinujoh — APPROVED** [review]
